configure.ac

```diff
@@ -841,12 +841,56 @@ AC_DEFINE_UNQUOTED([PACEMAKER_CONFIG_DIR], ["$PACEMAKER_CONFIG_DIR"],
                    [Where to keep configuration files like authkey])
 AC_SUBST(PACEMAKER_CONFIG_DIR)
 
-AC_DEFINE_UNQUOTED([SBIN_DIR], ["$sbindir"], [Location for system binaries])
+# Fedora >=42 makes /usr/sbin a symlink to /usr/bin. It updates the RPM macros
+# to set _sbindir to "${_exec_prefix}/bin", the same value as _bindir.
+# Previously it was set to "${_exec_prefix}/sbin". (Note that because of the
+# symlink, paths beginning with /usr/sbin remain valid.)
+#
+# This causes problems with bundle resources. Pacemaker automatically generates
+# a configuration for the bundle's container resource. If the bundle contains a
+# primitive and the container's run-command attribute is unset, the generated
+# container resource has its run_cmd attribute set to
+# SBIN_DIR "/" PCMK__SERVER_REMOTED, which is intended as a reasonable default.
+# If SBIN_DIR becomes "/usr/bin" instead of "/usr/sbin", at least two problems
+# can occur:
+#   1. The container resource's digest changes compared to the digest in the
+#      resource history entry. Pacemaker interprets this as a configuration
+#      change and restarts the container resource.
+#   2. If the container is running a different OS distro or an older version of
+#      Fedora, then the new /usr/bin/pacemaker-remoted path may be invalid; the
+#      executable was installed at /usr/sbin/pacemaker-remoted, which is NOT a
+#      symlink to /usr/bin path. In this case, the container fails to start.
+#
+# We override the value only for the SBIN_DIR constant, which is used only for
+# the sbd path and the default pacemaker-remoted path. There is no need to
+# override sbindir, which would affect installation directories.
+#
+# There is no more specific way than the below, to detect whether the build
+# system has this /usr/sbin vs. /usr/bin change in effect. Thus corner cases are
+# possible when sbindir/bindir are manually specified or in distros with
+# atypical defaults.
+#
+# At time of writing, autoconf is unchanged. However, we perform the override
+# here instead of in the spec file, in case autoconf changes in the future.
+#
+# Note that other distros (for example, RHEL) are likely to incorporate these
+# changes in the future.
+#
+# References:
+# * https://fedoraproject.org/wiki/Changes/Unify_bin_and_sbin
+# * https://bodhi.fedoraproject.org/updates/FEDORA-2025-da0a082e66
+# * https://discussion.fedoraproject.org/t/144562
+if test "$sbindir" == "$bindir" && test "$sbindir" == "${exec_prefix}/bin"; then
+    SBIN_DIR="${exec_prefix}/sbin"
+else
+    SBIN_DIR="$sbindir"
+fi
```

Comment:
I don't see how this could be the source of the rawhide CI failures, but this should be rewritten to use AS_IF for consistency with the rest of configure.ac.

Reply:
> how

2.1 RPM builds previously failed on `cts-cli` due to digest change (causing resource restarts) because the generated podman resource had a different `run_cmd` (`/usr/bin/pacemaker-remoted` vs. `/usr/sbin/pacemaker-remoted` when the digest in the resource history was generated). This is because on rawhide, the RPM build macros define `_sbindir` as `%{_exec_prefix}/bin` instead of `%{_exec_prefix}/sbin`. This then gets passed to `configure` as an override for `sbindir`. `configure` then defines `SBIN_DIR` to the value of `sbindir`.

With this commit, we override `SBIN_DIR` in this case... or in as specific a case as we can. This way, we avoid messing with `sbindir` itself and thus the installation directories.

> rewritten

Done.

Reply:
Right, I understand the bug and what the patch is trying to achieve.  What I meant was I don't see how using shell vs. AS_IF could possibly lead to CI failures.